### PR TITLE
Use Wandb experiment's files directory as save directory

### LINKED ISSUE #17654: Wandb checkpoints are not logged in experiment's directory
### Bug description

When using the Wandb Logger and setting the `log_model=True`, the model checkpoint isn't saved in the wandb experiment directory, but in the separate lightning logs directory.

Current behavior:
```
.
├── wandb
│   ├── run-20230517_120103-pgl4q0zq
│   │   ├── files
│   │   │   ├── config.yaml
│   │   │   ├── output.log
│   │   │   ├── ...
│   │   │   └── wandb-summary.json
│   │  ├── ...
│   │  └── logs
├── lightning_logs
|   └── pgl4q0zq
|       └── checkpoints
|           └── epoch=0-step=0.ckpt
```

Desired behavior:
```
.
.
├── wandb
│   ├── run-20230517_120103-pgl4q0zq
│   │   ├── files
│   │   │   ├── config.yaml
│   │   │   ├── output.log
│   │   │   ├── ...
│   │   │   ├── checkpoints
│   │   │   │   └── epoch=0-step=0.ckpt
│   │   │   └── wandb-summary.json
│   │  ├── ...
│   │  └── logs
```

### What version are you seeing the problem on?

master

### How to reproduce the bug

```python
from lightning import LightningModule, Trainer
from lightning.pytorch.loggers import WandbLogger
from torch.utils.data import Dataset, DataLoader
import torch


class MyDataModule(Dataset):
    def __getitem__(self, index) -> torch.Tensor:
        return torch.rand(1)

    def __len__(self) -> int:
        return 5


class MyLightningModule(LightningModule):
    def __init__(self):
        super().__init__()
        self.model = torch.nn.Linear(1, 1)

    def forward(self, x):
        return self.model(x)

    def training_step(self, batch, batch_idx):
        return self.model(batch).mean()

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters())


model = MyLightningModule()
data = MyDataModule()
loader = DataLoader(data, batch_size=1)
trainer = Trainer(logger=WandbLogger(log_model=True))
trainer.fit(model, loader)
```

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

### More info

I see two problems which need to be addressed:

1. The WandbLogger doesn't use the experiment's files directory as its `save_dir` property 
This can can fixed by replacing this [line](https://github.com/Lightning-AI/lightning/blob/deb70464967f4a996714874ebf44861c3481e0ff/src/lightning/pytorch/loggers/wandb.py#L493)) by `self.experiment.dir`

2. The ModelCheckpoint adds the name and version of the logger to the checkpoint path when resolving the checkpoint directory. The culprit is the [`__resolve_ckpt_dir`](https://github.com/Lightning-AI/lightning/blob/deb70464967f4a996714874ebf44861c3481e0ff/src/lightning/pytorch/callbacks/model_checkpoint.py#L580) function. I would propose moving the adding the name and version of the logger to the save_dir to the actual logger, as each logger may have a different strategy for integrating the version and name. For example, wandb does this by default and creates the files directory for saving artifacts.

### PR DESCRIPTION
**Pull request by fschlatt**
## What does this PR do?

Fixes #17654

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21462.org.readthedocs.build/en/21462/

### CODE REVIEW

**Review by fschlatt — COMMENTED** [review]
This PR is not ready to merge. I've reverted fixing the tests, because I misinterpreted what the `save_dir` argument did. Implementing proper tests is a bit more involved and is something I will try to solve soon.
